pylint/checkers/base/basic_checker.py

```diff
@@ -731,7 +772,9 @@ def leave_tryfinally(self, _: nodes.TryFinally) -> None:
         self._tryfinallys.pop()
 
     def _check_unreachable(
-        self, node: nodes.Return | nodes.Continue | nodes.Break | nodes.Raise
+        self,
+        node: nodes.Return | nodes.Continue | nodes.Break | nodes.Raise | nodes.Call,
+        confidence: Confidence = HIGH,
```

Comment:
Why do we need a default value here? Can't we just make this be required?

Reply:
Because in the terminating func case (what I've added in this PR), the confidence is INFERENCE, but in every other case, it's HIGH

Reply:
Let's make that explicit in all calling functions.

Reply:
Yeah, you're right. I should have checked. I thought the other calls to `_check_unreachable` included similar checks like `is_terminating` but they don't. They just directly call.

Never mind this comment, and good call on pushing back 😄